pkg/vm/cli/cli_test.go

```diff
@@ -0,0 +1,437 @@
+	e := &executor{
+		in:  &readCloser{Buffer: *bytes.NewBuffer(nil)},
+		out: bytes.NewBuffer(nil),
+		ch:  make(chan struct{}),
+	}
+	e.cli = NewWithConfig(&readline.Config{
+		Prompt: "",
+		Stdin:  e.in,
+		Stdout: e.out,
+	})
+	go func() {
+		require.NoError(t, e.cli.Run())
+		close(e.ch)
+	}()
+	return e
+}
+
+func (e *executor) runProg(t *testing.T, commands ...string) {
+	cmd := strings.Join(commands, "\n") + "\n"
+	e.in.WriteString(cmd + "\n")
+	select {
+	case <-e.ch:
+	case <-time.After(time.Second):
+		require.Fail(t, "command took too long time")
+	}
+}
+
+func (e *executor) checkNextLine(t *testing.T, expected string) {
+	line, err := e.out.ReadString('\n')
+	require.NoError(t, err)
+	require.Regexp(t, expected, line)
+}
+
+func (e *executor) checkError(t *testing.T, expectedErr error) {
+	line, err := e.out.ReadString('\n')
+	require.NoError(t, err)
+	require.True(t, strings.HasPrefix(line, "Error: "+expectedErr.Error()))
+}
+
+func (e *executor) checkStack(t *testing.T, items ...interface{}) {
+	d := json.NewDecoder(e.out)
+	var actual interface{}
+	require.NoError(t, d.Decode(&actual))
+	rawActual, err := json.Marshal(actual)
+	require.NoError(t, err)
+
+	expected := vm.NewStack("")
+	for i := range items {
+		expected.PushVal(items[i])
+	}
+	rawExpected, err := json.Marshal(expected)
+	require.NoError(t, err)
+	require.JSONEq(t, string(rawExpected), string(rawActual))
+
+	// Decoder has it's own buffer, we need to return unread part to the output.
+	outRemain := e.out.String()
+	e.out.Reset()
+	_, err = gio.Copy(e.out, d.Buffered())
+	require.NoError(t, err)
+	e.out.WriteString(outRemain)
+	_, err = e.out.ReadString('\n')
+	require.NoError(t, err)
+}
+
+func TestLoad(t *testing.T) {
+	script := []byte{byte(opcode.PUSH3), byte(opcode.PUSH4), byte(opcode.ADD)}
+	t.Run("loadhex", func(t *testing.T) {
+		e := newTestVMCLI(t)
+		e.runProg(t,
+			"loadhex",
+			"loadhex notahex",
+			"loadhex "+hex.EncodeToString(script))
+
+		e.checkError(t, ErrMissingParameter)
+		e.checkError(t, ErrInvalidParameter)
+		e.checkNextLine(t, "READY: loaded 3 instructions")
+	})
+	t.Run("loadbase64", func(t *testing.T) {
+		e := newTestVMCLI(t)
+		e.runProg(t,
+			"loadbase64",
+			"loadbase64 not_a_base64",
+			"loadbase64 "+base64.StdEncoding.EncodeToString(script))
+
+		e.checkError(t, ErrMissingParameter)
+		e.checkError(t, ErrInvalidParameter)
+		e.checkNextLine(t, "READY: loaded 3 instructions")
+	})
+
+	src := `package kek
+	func Main(op string, args []interface{}) int {
+		a := args[0].(int)
+		b := args[1].(int)
+		if op == "add" {
+			return a + b
+		} else {
+			return a * b
+		}
+	}`
+	t.Run("loadgo", func(t *testing.T) {
+		filename := path.Join(os.TempDir(), "vmtestcontract.go")
+		require.NoError(t, ioutil.WriteFile(filename, []byte(src), os.ModePerm))
+		filenameErr := path.Join(os.TempDir(), "vmtestcontract_err.go")
+		require.NoError(t, ioutil.WriteFile(filenameErr, []byte(src+"invalid_token"), os.ModePerm))
+		defer os.Remove(filename)
```

Comment:
It can be left behind in case of errors above. And better again use single temporary directory for all the stuff.

Reply:
Fixed.